Pear/Onboarding/ProfilePromptsViewController.swift

```diff
@@ -0,0 +1,207 @@
+        subtitleLabel.font = ._12CircularStdBook
+        subtitleLabel.textAlignment = .center
+        subtitleLabel.numberOfLines = 0
+        subtitleLabel.textColor = .darkGreen
+        view.addSubview(subtitleLabel)
+
+        fadeTableView.view.clipsToBounds = true
+        fadeTableView.view.backgroundColor = .clear
+        fadeTableView.view.keyboardDismissMode = .onDrag
+        fadeTableView.view.separatorStyle = .none
+        fadeTableView.view.contentInset = UIEdgeInsets(top: 5, left: 0, bottom: 50, right: 0)
+        fadeTableView.view.delegate = self
+        fadeTableView.view.rowHeight = UITableView.automaticDimension
+        fadeTableView.view.estimatedRowHeight = 140
+        fadeTableView.view.dataSource = self
+        fadeTableView.view.register(PromptTableViewCell.self, forCellReuseIdentifier: PromptTableViewCell.reuseIdentifier)
+        fadeTableView.view.separatorColor = .clear
+        view.addSubview(fadeTableView)
+
+        nextButton.setTitle("Next", for: .normal)
+        nextButton.setTitleColor(.white, for: .normal)
+        nextButton.titleLabel?.font = ._20CircularStdBold
+        nextButton.backgroundColor = .inactiveGreen
+        nextButton.layer.cornerRadius = 26
+        nextButton.isEnabled = false
+        nextButton.addTarget(self, action: #selector(nextButtonPressed), for: .touchUpInside)
+        view.addSubview(nextButton)
+
+        setupConstraints()
+    }
+
+    private func updateNext() {
+        nextButton.isEnabled = prompts.filter{ $0.didAnswerPrompt }.count == 3
+        nextButton.backgroundColor = nextButton.isEnabled ? .backgroundOrange : .inactiveGreen
+    }
+
+    @objc private func backButtonPressed() {
+        delegate?.backPage(index: 2)
+    }
+
+    @objc private func nextButtonPressed() {
+        // TODO - update prompt values in backend
+        self.delegate?.nextPage(index: 4)
+    }
+
+    private func setupConstraints() {
+
+        titleLabel.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide).offset(Constants.Onboarding.titleLabelPadding)
+            make.height.equalTo(61)
+            make.width.equalTo(295)
+            make.centerX.equalToSuperview()
+        }
+
+        backButton.snp.makeConstraints { make in
+            make.centerY.equalTo(titleLabel)
+            make.leading.equalToSuperview().offset(24)
+            make.size.equalTo(Constants.Onboarding.backButtonSize)
+        }
+
+        subtitleLabel.snp.makeConstraints { make in
+            make.width.equalTo(titleLabel)
+            make.height.equalTo(34)
+            make.top.equalTo(titleLabel.snp.bottom).offset(12)
+            make.centerX.equalToSuperview()
+        }
+
+        fadeTableView.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.leading.trailing.equalToSuperview().inset(48)
+            make.bottom.equalTo(nextButton.snp.top).offset(-57)
```

Comment:
Does setting all the `edge` constraints not give you constraint errors with your `centerX` constraint?

Reply:
oh it didn't, but I see that it's redundant so I'll get rid of the centerX constraint